pkg/controllers/nodepool/status/controller.go

```diff
@@ -0,0 +1,134 @@
+/*
+Copyright The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package status
+
+import (
+	"context"
+	"fmt"
+
+	"go.uber.org/multierr"
+	"k8s.io/apimachinery/pkg/api/errors"
+
+	"sigs.k8s.io/karpenter/pkg/utils/nodepool"
+
+	"github.com/awslabs/operatorpkg/status"
+	"k8s.io/apimachinery/pkg/api/equality"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"knative.dev/pkg/logging"
+
+	"sigs.k8s.io/karpenter/pkg/cloudprovider"
+	"sigs.k8s.io/karpenter/pkg/operator/injection"
+
+	controllerruntime "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	"sigs.k8s.io/karpenter/pkg/apis/v1beta1"
+)
+
+// Controller for the resource
+type Controller struct {
+	kubeClient    client.Client
+	cloudProvider cloudprovider.CloudProvider
+}
+
+// NewController is a constructor
+func NewController(kubeClient client.Client, cloudProvider cloudprovider.CloudProvider) *Controller {
+	return &Controller{
+		kubeClient:    kubeClient,
+		cloudProvider: cloudProvider,
+	}
+}
+
+func (c *Controller) Reconcile(ctx context.Context, nodePool *v1beta1.NodePool) (reconcile.Result, error) {
+	ctx = logging.WithLogger(ctx, logging.FromContext(ctx).Named("nodepool.status").With("nodepool", nodePool.Name))
+	ctx = injection.WithControllerName(ctx, "nodepool.status")
+	stored := nodePool.DeepCopy()
+
+	nodeClass := &unstructured.Unstructured{}
+	nodeClass.SetAPIVersion(nodePool.Spec.Template.Spec.NodeClassRef.APIVersion)
+	nodeClass.SetKind(nodePool.Spec.Template.Spec.NodeClassRef.Kind)
+	supportedNC := c.cloudProvider.GetSupportedNodeClasses()
+	if len(nodeClass.GetAPIVersion()) == 0 {
+		nodeClass.SetAPIVersion(fmt.Sprintf("%s/%s", supportedNC[0].Group, supportedNC[0].Version))
+	}
+	if len(nodeClass.GetKind()) == 0 {
+		nodeClass.SetKind(supportedNC[0].Kind)
+	}
+	if err := c.kubeClient.Get(ctx, client.ObjectKey{Name: nodePool.Spec.Template.Spec.NodeClassRef.Name}, nodeClass); err != nil {
+		return reconcile.Result{}, fmt.Errorf("resolving nodeClass, %w", err)
+	}
+	ready, err := c.IsReady(status.ConditionReady, nodeClass)
+	if err != nil || !ready {
+		nodePool.StatusConditions().SetFalse(v1beta1.ConditionTypeNodeClassReady, "NodeClassNotReady", "unable to resolve underlying nodeClass")
```

Comment:
```suggestion
		nodePool.StatusConditions().SetFalse(v1beta1.ConditionTypeNodeClassReady, "NodeClassNotReady", "Unable to resolve underlying nodeClass")
```

nit: Messages should always start with a capital letter. Same thing with reasons, but you've already got that 👍🏼